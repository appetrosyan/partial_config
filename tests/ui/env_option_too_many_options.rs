#[derive(partial_config::HasPartial, partial_config::EnvSourced)]
pub struct Configuration {
    #[env(OPTIONAL)]
    pub optional: Option<Option<String>>
}

fn main()  {}
