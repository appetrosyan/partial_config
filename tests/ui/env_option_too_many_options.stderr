error[E0277]: the trait bound `Option<String>: FromStr` is not satisfied
 --> tests/ui/env_option_too_many_options.rs:4:26
  |
4 |     pub optional: Option<Option<String>>
  |                          ^^^^^^^^^^^^^^ the trait `FromStr` is not implemented for `Option<String>`
  |
  = help: the following other types implement trait `FromStr`:
            IpAddr
            Ipv4Addr
            Ipv6Addr
            NonZero<i128>
            NonZero<i16>
            NonZero<i32>
            NonZero<i64>
            NonZero<i8>
          and $N others

error[E0277]: the trait bound `Option<String>: FromStr` is not satisfied
 --> tests/ui/env_option_too_many_options.rs:1:38
  |
1 | #[derive(partial_config::HasPartial, partial_config::EnvSourced)]
  |                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `FromStr` is not implemented for `Option<String>`
  |
  = help: the following other types implement trait `FromStr`:
            IpAddr
            Ipv4Addr
            Ipv6Addr
            NonZero<i128>
            NonZero<i16>
            NonZero<i32>
            NonZero<i64>
            NonZero<i8>
          and $N others
  = note: this error originates in the derive macro `partial_config::EnvSourced` (in Nightly builds, run with -Z macro-backtrace for more info)
